pyanaconda/modules/subscription/subscription.py

```diff
@@ -0,0 +1,235 @@
+        - there is no list of valid addons
+
+        Due to this we need to take into account that the listing
+        might not always be comprehensive and that we need to allow
+        what might on a first glance look like invalid values to
+        be written to the target system.
+        """
+        roles, slas, usage_types = system_purpose.get_valid_fields()
+        self._valid_roles = roles
+        self._valid_slas = slas
+        self._valid_usage_types = usage_types
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(SUBSCRIPTION.object_path, SubscriptionInterface(self))
+        DBus.register_service(SUBSCRIPTION.service_name)
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return SubscriptionKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        log.debug("Processing kickstart data...")
+        # Try if any of the values in kickstart match a valid field.
+        # If it does, write the valid field value instead of the value
+        # from kickstart.
+        # This way a value in kickstart that has a different case
+        # and/or trailing white space can still be used to preselect a value in
+        # a UI instead of being marked as a custom user specified value.
+        if data.syspurpose.role:
+            role_match = system_purpose.match_field(data.syspurpose.role, self.valid_roles)
+        else:
+            role_match = None
+
+        if data.syspurpose.sla:
+            sla_match = system_purpose.match_field(data.syspurpose.sla, self.valid_slas)
+        else:
+            sla_match = None
+
+        if data.syspurpose.usage:
+            usage_match = system_purpose.match_field(data.syspurpose.usage, self._valid_usage_types)
+        else:
+            usage_match = None
+
+        if role_match:
+            log.info("role value %s from kickstart matched to know valid field %s", data.syspurpose.role, role_match)
+            self.set_role(role_match)
+        elif data.syspurpose.role:
+            log.info("using custom role value from kickstart: %s", data.syspurpose.role)
+            self.set_role(data.syspurpose.role)
+
+        if sla_match:
+            log.info("SLA value %s from kickstart matched to know valid field %s", data.syspurpose.sla, sla_match)
+            self.set_sla(sla_match)
+        elif data.syspurpose.sla:
+            log.info("using custom SLA value from kickstart: %s", data.syspurpose.sla)
+            self.set_sla(data.syspurpose.sla)
+
+        if usage_match:
+            log.info("usage value %s from kickstart matched to know valid field %s", data.syspurpose.usage, usage_match)
+            self.set_usage(usage_match)
+        elif data.syspurpose.usage:
+            log.info("using custom usage value from kickstart: %s", data.syspurpose.usage)
+            self.set_usage(data.syspurpose.usage)
```

Comment:
These are basically 3 independent parts. Could you please split them to 3 independent methods. They are share anything so or so.